mojo/stdlib/stdlib/os/process.mojo

```diff
@@ -0,0 +1,290 @@
+        """
+
+        self.child_pid = child_pid
+
+    fn _kill(self, signal: Int) -> Bool:
+        # `kill` returns 0 on success and -1 on failure
+        return kill(self.child_pid, signal) > -1
+
+    fn hangup(self) -> Bool:
+        """Send the Hang up signal to the managed child process.
+
+        Returns:
+          Upon successful completion, True is returned else False.
+        """
+        return self._kill(SignalCodes.HUP)
+
+    fn interrupt(self) -> Bool:
+        """Send the Interrupt signal to the managed child process.
+
+        Returns:
+          Upon successful completion, True is returned else False.
+        """
+        return self._kill(SignalCodes.INT)
+
+    fn kill(self) -> Bool:
+        """Send the Kill signal to the managed child process.
+
+        Returns:
+          Upon successful completion, True is returned else False.
+        """
+        return self._kill(SignalCodes.KILL)
+
+    @staticmethod
+    fn run(var path: String, argv: List[String]) raises -> Process:
+        """Spawn new process from file executable.
+
+        Args:
+          path: The path to the file.
+          argv: A list of string arguments to be passed to executable.
+
+        Returns:
+          An instance of `Process` struct.
+        """
+
+        @parameter
+        if CompilationTarget.is_linux() or CompilationTarget.is_macos():
+            var file_name = String(path.split(sep)[-1])
+            var pipe = Pipe(out_close_on_exec=True)
+            var exec_err_code = StaticString("EXEC_ERR")
+
+            var pid = vfork()
+
+            if pid == 0:
+                # Child process.
+                pipe.set_output_only()
+
+                var arg_count = len(argv)
+                var argv_array_ptr_cstr_ptr = UnsafePointer[
+                    UnsafePointer[c_char, mut=False]
+                ].alloc(arg_count + 2)
+                var offset = 0
+                # Arg 0 in `argv` ptr array should be the file name
+                argv_array_ptr_cstr_ptr[offset] = file_name.unsafe_cstr_ptr()
+                offset += 1
+
+                for var arg in argv:
+                    argv_array_ptr_cstr_ptr[offset] = arg.unsafe_cstr_ptr()
+                    offset += 1
+
+                # `argv` ptr array terminates with NULL PTR
+                argv_array_ptr_cstr_ptr[offset] = UnsafePointer[c_char]()
+
+                var path_cptr = path.unsafe_cstr_ptr()
+
+                _ = execvp(path_cptr, argv_array_ptr_cstr_ptr)
+
+                # This will only get reached if exec call fails to replace currently executing code
+                argv_array_ptr_cstr_ptr.free()
+
+                # Canonical fork/ exec error handling pattern of using a pipe that closes on exec is
+                # used to signal error to parent process `https://cr.yp.to/docs/selfpipe.html`
+                pipe.write_bytes(exec_err_code.as_bytes())
+
+                exit(1)
```

Comment:
I think this will call the parent's atexit handlers and flush the parents io buffers